### PR TITLE
A handy iterator for experiments

### PR DESCRIPTION
**Pull request by graeme-winter**
imageset = experiment.imageset
    for expt in imageset.as_experiments():
        print(expt.is_still(), expt.scan.get_array_range())

To allow easy iteration through ImageSequences for e.g. indexing

### CODE REVIEW

File: imageset.py [hunk 1]
```diff
@@ -234,6 +234,25 @@ def __getitem__(self, item):
         else:
             return self.get_corrected_data(item)
 
+    def as_experiments(self):
+        """Iterate over images in the sequence creating an experiment for
+        each with the information in the ImageSequence."""
+
+        start, end = self.get_scan().get_array_range()
+
+        from dxtbx.model.experiment_list import Experiment
```
Comment (dagewa) [inline imageset.py]: Import should probably be floated

**Comment by rjgildea** [comment]
It isn't quite clear to me what the use case for this is. As far as I can tell from examining the source code (since there aren't any tests), it appears that given a sweep as input, you would get one experiment per image as output, which isn't necessarily what I would expect from `imageset.as_experiments()`.

I haven't tested the code below, but it is my understanding of the behaviour from looking at the code:
```
>>> len(experiments)
1
>>> imageset = experiments[0].imageset
>>> expts = ExperimentList(imageset.as_experiments())
>>> assert len(expts) == len(imagesets)
>>> assert len(expts) != len(experiments)
```
This seems like potentially confusing behaviour.

What happens if you give an imageset that doesn't have a scan set?

**Comment by graeme-winter** [comment]
Use case is for stills - wanting to generate some useful sets of data for working on beyond spot finding hence adding to image sequence. Could overload in image sweep to give exactly one experiment, but for stills we would expect each image to be an independent experiment. 

This is draft for a reason ;-)

File: imageset.py (hunk 1 above)
Comment (graeme-winter) [reply imageset.py]: I was concerned about circular imports for this...

Comment (dagewa) [reply imageset.py]: I don't think that matters, and in this case you are just importing `Experiment`, which is a boost python object

Comment (graeme-winter) [reply imageset.py]: Seems to have gone away

**Comment by graeme-winter** [comment]
Oh I think this is the dead one, sorry

**Pull request closed without merging**